awswrangler/lakeformation/_read.py

```diff
@@ -0,0 +1,192 @@
+"""Amazon Lake Formation Module gathering all read functions."""
+import concurrent.futures
+import itertools
+import logging
+from typing import Any, Dict, Iterator, List, Optional, Tuple, Union
+
+import boto3
+import pandas as pd
+from pyarrow import NativeFile, RecordBatchStreamReader
+
+from awswrangler import _utils, exceptions
+from awswrangler._config import apply_configs
+from awswrangler.lakeformation._utils import wait_query
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _execute_query(
+    query_id: str,
+    token_work_unit: Tuple[str, int],
+    boto3_session: Optional[boto3.Session] = None,
+) -> pd.DataFrame:
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=boto3_session)
+    token, work_unit = token_work_unit
+    messages: NativeFile = client_lakeformation.execute(QueryId=query_id, Token=token, WorkUnitId=work_unit)["Messages"]
+    return RecordBatchStreamReader(messages.read()).read_pandas()
```

Comment:
Here I think we should consider all possible arguments like [here](https://github.com/awslabs/aws-data-wrangler/blob/54a266c860a1a9a6d4ec31aaaf6f22ccf92c1b5c/awswrangler/s3/_read_parquet.py#L250), mainly `types_mapper`.

[Reference](https://arrow.apache.org/docs/python/generated/pyarrow.ipc.RecordBatchFileReader.html?highlight=read_pandas#pyarrow.ipc.RecordBatchFileReader.read_pandas)

Reply:
Good point, passing the same arguments in new commit